## 单利与复利

- 基本概念
    - FV：future value，未来值
    - PV：present value，现值
    - interest：利率
- 单利与复利计算
    - 利率 $r$，期数 $N$
    $$
    \begin{split}
    &FV=PV\cdot(1+r\cdot N)\\
    &FV=PV\cdot(1+r)^N
    \end{split}
    $$
    

### 计算示例

- PV：10000，年化利率 6%, 20 年；
- 单利：
    $$
    10000\left(1+0.06\cdot 20\right)=22000
    $$
- 复利（按年计息，一年记一次息）
    $$
    10000\left(1+\frac{0.06}{1}\right)^{20}=32071
    $$
- 复利（按月计息）

    $$
    10000\left(1+\frac{0.06}{12}\right)^{20\times 12}=33102
    $$

- 复利（按天计息）

    $$
    10000\left(1+\frac{0.06}{365}\right)^{20\times 365}=33198
    $$

## 72法则

In [3]:
import math
math.log(2)

0.6931471805599453

- 给定利率水平 $r$ 下，求 FV = 2*PV 时（本息和翻倍）所需的期数 $N$；

$$
\begin{split}
FV&=2PV=PV(1+r)^N\\
N&=\frac{\ln 2}{\ln (1+r)}\approx \frac{72\%}{r}
\end{split}
$$

- $\ln 2\approx 0.693\approx 0.72$
    - 72 的约数更多，更容易做快速计算；
- $\ln (1+r)=r-\frac{r^2}2+\frac{r^3}3+\cdots\approx r$

In [8]:
# r = 0.06 时，需要12年
print((1+0.06)**12)
# r = 0.08 时，需要 9 年
print((1+0.08)**9)
# r = 0.12 时，需要 6年
print((1+0.12)**6)

2.0121964718355514
1.9990046271044333
1.973822685184001
